crypto/ec/ec_asn1.c

```diff
@@ -15,15 +15,18 @@
 
 int EC_GROUP_get_basis_type(const EC_GROUP *group)
 {
-    int i = 0;
+    int i;
 
     if (EC_METHOD_get_field_type(EC_GROUP_method_of(group)) !=
         NID_X9_62_characteristic_two_field)
         /* everything else is currently not supported */
         return 0;
 
-    while (group->poly[i] != 0)
-        i++;
+    /* Find the last non-zero element of group->poly[] */
+    for (i = 0;
+         i < OSSL_NELEM(group->poly) & group->poly[i] != 0;
```

Comment:
Damnit, didn't see this before...  Surely, the lonely `&` is supposed to have some company, i.e. `&&`